ext/intl/listformatter/listformatter_class.c

```diff
@@ -0,0 +1,236 @@
+/*
+   +----------------------------------------------------------------------+
+   | This source file is subject to version 3.01 of the PHP license,      |
+   | that is bundled with this package in the file LICENSE, and is        |
+   | available through the world-wide-web at the following url:           |
+   | https://www.php.net/license/3_01.txt                                 |
+   | If you did not receive a copy of the PHP license and are unable to   |
+   | obtain it through the world-wide-web, please send a note to          |
+   | license@php.net so we can mail you a copy immediately.               |
+   +----------------------------------------------------------------------+
+   | Authors: Bogdan Ungureanu <bogdanungureanu21@gmail.com>              |
+   +----------------------------------------------------------------------+
+*/
+
+#include "php.h"
+#include "php_intl.h"
+#include <unicode/ulistformatter.h>
+#include "listformatter_arginfo.h"
+#include "listformatter_class.h"
+#include "intl_convert.h"
+
+static zend_object_handlers listformatter_handlers;
+
+static void listformatter_free_obj(zend_object *object)
+{
+    ListFormatter_object *obj = php_intl_listformatter_fetch_object(object);
+
+    if( obj->lf_data.ulistfmt )
+        ulistfmt_close( obj->lf_data.ulistfmt );
+
+    obj->lf_data.ulistfmt = NULL;
+    intl_error_reset( &obj->lf_data.error );
+
+    zend_object_std_dtor(&obj->zo);
+}
+
+static zend_object *listformatter_create_object(zend_class_entry *class_type)
+{
+    ListFormatter_object *obj;
+    obj = zend_object_alloc(sizeof(ListFormatter_object), class_type);
+
+    obj->lf_data.ulistfmt = NULL;
+    intl_error_reset( &obj->lf_data.error );
+
+    zend_object_std_init(&obj->zo, class_type);
+    object_properties_init(&obj->zo, class_type);
+    obj->zo.handlers = &listformatter_handlers;
+    return &obj->zo;
+}
+
+PHP_METHOD(IntlListFormatter, __construct)
+{
+    ListFormatter_object *obj = Z_INTL_LISTFORMATTER_P(ZEND_THIS);
+    char* locale;
+    size_t locale_len = 0;
+    zend_long type = ULISTFMT_TYPE_AND;
+    zend_long width = ULISTFMT_WIDTH_WIDE;
+    ZEND_PARSE_PARAMETERS_START(1, 3)
+        Z_PARAM_STRING(locale, locale_len)
+        Z_PARAM_OPTIONAL
+        Z_PARAM_LONG(type)
+        Z_PARAM_LONG(width)
+    ZEND_PARSE_PARAMETERS_END();
+
+    if(locale_len == 0) {
+        locale = (char *)intl_locale_get_default();
+    }
+
+    if (locale_len > INTL_MAX_LOCALE_LEN) {
+        zend_argument_value_error(1, "Locale string too long, should be no longer than %d characters", INTL_MAX_LOCALE_LEN);
```

Comment:
```suggestion
        zend_argument_value_error(1, "must be less than or equal to %d characters", INTL_MAX_LOCALE_LEN);
```

(the new error message is negotiable, but the current message is slightly off)

Reply:
@kocsismate I've opened https://github.com/php/php-src/pull/19282 for it.